influxql/ast.go

```diff
@@ -137,6 +135,13 @@ type Statement interface {
 	RequiredPrivileges() ExecutionPrivileges
 }
 
+// HasDefaultDB provides an interface to get the default DB from a Statement.
+type HasDefaultDB interface {
```

Comment:
I'll change it to `Database`.  I know `er` is idiomatic for interface names in Go but when it results in something like `DatabaseDefaulter` or `DefaultDatabaser`, it's time to break the rule.  :)  All statements could have a method that returns a default database or nil but it's simpler to only implement it for the few statements that need it.  Most statements won't need it, so it'll just 20+ funcs returning `nil` to ast.go.